apoc.export.cypher.query bug on multiple node index

## Expected Behavior (Mandatory)
To export given query to file with cypher statements.

## Actual Behavior (Mandatory)
Error:
```Neo.ClientError.Procedure.ProcedureCallFailed: Failed to invoke procedure `apoc.export.cypher.query`: Caused by: java.lang.IllegalStateException: This is a multi-token index, which has more than one label. Call the getLabels() method instead.```

## How to Reproduce the Problem
Create 

### Simple Dataset (where it's possibile)

```
CREATE (n:TempNode {value:"value"});
CREATE (n:TempNode2 {value:"value"});
CALL db.index.fulltext.createNodeIndex("MyCoolFulltextIndex",["TempNode", "TempNode2"],["value"]);
```


### Steps (Mandatory)

  1. Seed the DB with the dataset above
  2. Run `call apoc.export.cypher.query( "MATCH (t:TempNode) return t", "awesomeFile.cypher")`

## Screenshots (where it's possibile)

## Specifications (Mandatory)

Currently used versions

### Versions

  - OS: MacOS 10.14.6 (18G87)
  - Neo4j: 3.5.6 enterprise
  - Neo4j-Apoc: 3.5.0.4

--- a/docs/asciidoc/export/exportCypher.adoc
+++ b/docs/asciidoc/export/exportCypher.adoc
@@ -55,6 +55,9 @@ To change the cypher update operation, you have to set it on the config params l
 * `addStructure`: MATCH for nodes + MERGE for rels
 * `updateStructure`: MERGE + MATCH for nodes and rels
 
+In case of `cypher-shell` export format you can also set the timeout of `db.awaitIndexes` with the `awaitIndexes`
+configuration param.
+
 Format and cypherFormat can be used both in the same query giving you complete control over the exact export format:
 
 [source,cypher]

--- a/src/main/java/apoc/export/cypher/ExportCypher.java
+++ b/src/main/java/apoc/export/cypher/ExportCypher.java
@@ -10,15 +10,14 @@ import apoc.util.Util;
 import org.neo4j.cypher.export.CypherResultSubGraph;
 import org.neo4j.cypher.export.DatabaseSubGraph;
 import org.neo4j.cypher.export.SubGraph;
-import org.neo4j.graphdb.GraphDatabaseService;
-import org.neo4j.graphdb.Node;
-import org.neo4j.graphdb.Relationship;
-import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.*;
 import org.neo4j.helpers.collection.Iterables;
 import org.neo4j.procedure.*;
 
 import java.io.IOException;
-import java.util.*;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
 import java.util.concurrent.ArrayBlockingQueue;
 import java.util.concurrent.BlockingQueue;
 import java.util.stream.Stream;
@@ -77,7 +76,12 @@ public class ExportCypher {
         if (Util.isNullOrEmpty(fileName)) fileName=null;
         ExportConfig c = new ExportConfig(config);
         Result result = db.execute(query);
-        SubGraph graph = CypherResultSubGraph.from(result, db, c.getRelsInBetween());
+        SubGraph graph;
+        try {
+            graph = CypherResultSubGraph.from(result, db, c.getRelsInBetween());
+        } catch (IllegalStateException e) {
+            throw new RuntimeException("Full-text indexes on relationships are not supported, please delete them in order to complete the process");
+        }
         String source = String.format("statement: nodes(%d), rels(%d)",
                 Iterables.count(graph.getNodes()), Iterables.count(graph.getRelationships()));
         return exportCypher(fileName, source, graph, c, false);

--- a/src/main/java/apoc/export/cypher/MultiStatementCypherSubGraphExporter.java
+++ b/src/main/java/apoc/export/cypher/MultiStatementCypherSubGraphExporter.java
@@ -6,11 +6,9 @@ import apoc.export.util.ExportConfig;
 import apoc.export.util.ExportFormat;
 import apoc.export.util.Reporter;
 import apoc.util.Util;
+import org.apache.commons.lang3.StringUtils;
 import org.neo4j.cypher.export.SubGraph;
-import org.neo4j.graphdb.GraphDatabaseService;
-import org.neo4j.graphdb.Label;
-import org.neo4j.graphdb.Node;
-import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.*;
 import org.neo4j.graphdb.schema.IndexDefinition;
 import org.neo4j.helpers.collection.Iterables;
 
@@ -18,6 +16,7 @@ import java.io.IOException;
 import java.io.PrintWriter;
 import java.util.*;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
 import static apoc.export.cypher.formatter.CypherFormatterUtils.UNIQUE_ID_LABEL;
@@ -31,6 +30,28 @@ import static apoc.export.cypher.formatter.CypherFormatterUtils.UNIQUE_ID_PROP;
  * Outputs indexes and constraints at the beginning as their own transactions
  */
 public class MultiStatementCypherSubGraphExporter {
+
+    private enum IndexType {
+        NODE_LABEL_PROPERTY("node_label_property"),
+        NODE_UNIQUE_PROPERTY("node_unique_property"),
+        REL_TYPE_PROPERTY("relationship_type_property"),
+        NODE_FULLTEXT("node_fulltext"),
+        RELATIONSHIP_FULLTEXT("relationship_fulltext");
+
+        private final String typeName;
+
+        IndexType(String typeName) {
+            this.typeName = typeName;
+        }
+
+        static IndexType from(String stringType) {
+            return Stream.of(IndexType.values()).filter(type -> type.typeName().equals(stringType)).findFirst().orElse(null);
+        }
+
+        public String typeName() {
+            return typeName;
+        }
+    }
 
     private final SubGraph            graph;
     private final Map<String, Set<String>> uniqueConstraints = new HashMap<>();
@@ -170,10 +191,12 @@ public class MultiStatementCypherSubGraphExporter {
     // ---- Schema ----
 
     private void exportSchema(PrintWriter out) {
-        List<String> indexes = exportIndexes();
-        if (indexes.isEmpty() && artificialUniques == 0) return;
+        List<String> indexesAndConstraints = new ArrayList<>();
+        indexesAndConstraints.addAll(exportIndexes());
+        indexesAndConstraints.addAll(exportConstraints());
+        if (indexesAndConstraints.isEmpty() && artificialUniques == 0) return;
         begin(out);
-        for (String index : indexes) {
+        for (String index : indexesAndConstraints) {
             out.println(index);
         }
         if (artificialUniques > 0) {
@@ -183,43 +206,84 @@ public class MultiStatementCypherSubGraphExporter {
             }
         }
         commit(out);
-        List<String> indexesAwait = indexesAwait();
-        for (String indexAwait : indexesAwait) {
-            out.print(indexAwait);
+        if (graph.getIndexes().iterator().hasNext()) {
+            out.print(this.exportFormat.indexAwait(this.exportConfig.getAwaitForIndexes()));
         }
         schemaAwait(out);
         out.flush();
     }
 
     private List<String> exportIndexes() {
-        List<String> result = new ArrayList<>();
-        for (IndexDefinition index : graph.getIndexes()) {
-            String label = index.getLabel().name();
-            Iterable<String> props = index.getPropertyKeys();
-            if (index.isConstraintIndex()) {
-                String cypher = this.cypherFormat.statementForConstraint(label, props);
-                if (cypher != null && !"".equals(cypher)) {
-                    result.add(cypher);
-                }
-            } else {
-                String cypher = this.cypherFormat.statementForIndex(label, props);
-                if (cypher != null && !"".equals(cypher)) {
-                    result.add(0, cypher);
-                }
-            }
-        }
-        return result;
+        Result execute = db.execute("CALL db.indexes()");
+
+        return execute.stream()
+                .map(map -> {
+                    List<String> props = (List<String>) map.get("properties");
+                    List<String> tokenNames = (List<String>) map.get("tokenNames");
+                    String name = (String) map.get("indexName");
+                    boolean inGraph = tokensInGraph(tokenNames);
+                    if (!inGraph) {
+                        return null;
+                    }
+                    switch (IndexType.from(map.get("type").toString())) {
+                        case RELATIONSHIP_FULLTEXT:
+                            List<RelationshipType> types = toRelationshipTypes(tokenNames);
+                            return this.cypherFormat.statementForRelationshipFullTextIndex(name,
+                                    types, props);
+                        case NODE_FULLTEXT:
+                            List<Label> labels = toLabels(tokenNames);
+                            return this.cypherFormat.statementForNodeFullTextIndex(name,
+                                    labels, props);
+                        case NODE_UNIQUE_PROPERTY:
+                            return null; // delegate to the constraint creation
+                        default:
+                            String tokenName = tokenNames.get(0);
+                            return this.cypherFormat.statementForIndex(tokenName, props);
+                    }
+                })
+                .filter(StringUtils::isNotBlank)
+                .collect(Collectors.toList());
     }
 
-    private List<String> indexesAwait() {
-        List<String> result = new ArrayList<>();
-        for (IndexDefinition index : graph.getIndexes()) {
-            String label = index.getLabel().name();
-            String indexAwait = this.exportFormat.indexAwait(label, index.getPropertyKeys());
-            if (indexAwait != null && !"".equals(indexAwait))
-                result.add(indexAwait);
-        }
-        return result;
+    private boolean tokensInGraph(List<String> tokens) {
+        return StreamSupport.stream(graph.getIndexes().spliterator(), false)
+                .anyMatch(indexDefinition -> {
+                    if (indexDefinition.isRelationshipIndex()) {
+                        List<String> typeNames = StreamSupport.stream(indexDefinition.getRelationshipTypes().spliterator(), false)
+                                .map(RelationshipType::name)
+                                .collect(Collectors.toList());
+                        return typeNames.containsAll(tokens);
+                    } else {
+                        List<String> labelNames = StreamSupport.stream(indexDefinition.getLabels().spliterator(), false)
+                                .map(Label::name)
+                                .collect(Collectors.toList());
+                        return labelNames.containsAll(tokens);
+                    }
+                });
+    }
+
+    private List<Label> toLabels(List<String> tokenNames) {
+        return tokenNames.stream()
+                .map(Label::label)
+                .collect(Collectors.toList());
+    }
+
+    private List<RelationshipType> toRelationshipTypes(List<String> tokenNames) {
+        return tokenNames.stream()
+                .map(RelationshipType::withName)
+                .collect(Collectors.toList());
+    }
+
+    private List<String> exportConstraints() {
+        return StreamSupport.stream(graph.getIndexes().spliterator(), false)
+                .filter(index -> index.isConstraintIndex())
+                .map(index -> {
+                    String label = index.getLabel().name();
+                    Iterable<String> props = index.getPropertyKeys();
+                    return this.cypherFormat.statementForConstraint(label, props);
+                })
+                .filter(StringUtils::isNotBlank)
+                .collect(Collectors.toList());
     }
 
     // ---- CleanUp ----
@@ -265,15 +329,17 @@ public class MultiStatementCypherSubGraphExporter {
     }
 
     private void gatherUniqueConstraints() {
-        for (IndexDefinition indexDefinition : graph.getIndexes()) {
-            String label = indexDefinition.getLabel().name();
-            Set<String> props = StreamSupport
-                    .stream(indexDefinition.getPropertyKeys().spliterator(), false)
+        for (IndexDefinition index : graph.getIndexes()) {
+            Set<String> label = StreamSupport.stream(index.getLabels().spliterator(), false)
+                    .map(l -> l.name())
                     .collect(Collectors.toSet());
-            indexNames.add(label);
+            Set<String> props = StreamSupport
+                    .stream(index.getPropertyKeys().spliterator(), false)
+                    .collect(Collectors.toSet());
+            indexNames.add(index.getName());
             indexedProperties.addAll(props);
-            if (indexDefinition.isConstraintIndex()) { // we use the constraint that have few properties
-                uniqueConstraints.compute(label, (k, v) ->  v == null || v.size() > props.size() ? props : v);
+            if (index.isConstraintIndex()) { // we use the constraint that have few properties
+                uniqueConstraints.compute(String.join(":", label), (k, v) ->  v == null || v.size() > props.size() ? props : v);
             }
         }
     }

--- a/src/main/java/apoc/export/cypher/formatter/AbstractCypherFormatter.java
+++ b/src/main/java/apoc/export/cypher/formatter/AbstractCypherFormatter.java
@@ -29,6 +29,10 @@ abstract class AbstractCypherFormatter implements CypherFormatter {
 
 	private static final String STATEMENT_CONSTRAINTS = "CREATE CONSTRAINT ON (node:%s) ASSERT (%s) %s;";
 
+	private static final String STATEMENT_NODE_FULLTEXT_IDX = "CALL db.index.fulltext.createNodeIndex('%s',[%s],[%s]);";
+	private static final String STATEMENT_REL_FULLTEXT_IDX = "CALL db.index.fulltext.createRelationshipIndex('%s',[%s],[%s]);";
+	public static final String PROPERTY_QUOTING_FORMAT = "'%s'";
+
 	@Override
 	public String statementForCleanUp(int batchSize) {
 		return "MATCH (n:" + Q_UNIQUE_ID_LABEL + ") " +
@@ -39,6 +43,34 @@ abstract class AbstractCypherFormatter implements CypherFormatter {
 	@Override
 	public String statementForIndex(String label, Iterable<String> keys) {
 		return "CREATE INDEX ON :" + Util.quote(label) + "(" + CypherFormatterUtils.quote(keys) + ");";
+	}
+
+	@Override
+	public String statementForNodeFullTextIndex(String name, Iterable<Label> labels, Iterable<String> keys) {
+		String label = StreamSupport.stream(labels.spliterator(), false)
+				.map(Label::name)
+				.map(Util::quote)
+				.map(s -> String.format(PROPERTY_QUOTING_FORMAT, s))
+				.collect(Collectors.joining(","));
+		String key = StreamSupport.stream(keys.spliterator(), false)
+				.map(Util::quote)
+				.map(s -> String.format(PROPERTY_QUOTING_FORMAT, s))
+				.collect(Collectors.joining(","));
+		return String.format(STATEMENT_NODE_FULLTEXT_IDX, name, label, key);
+	}
+
+	@Override
+	public String statementForRelationshipFullTextIndex(String name, Iterable<RelationshipType> types, Iterable<String> keys) {
+		String type = StreamSupport.stream(types.spliterator(), false)
+				.map(RelationshipType::name)
+				.map(Util::quote)
+				.map(s -> String.format(PROPERTY_QUOTING_FORMAT, s))
+				.collect(Collectors.joining(","));
+		String key = StreamSupport.stream(keys.spliterator(), false)
+				.map(Util::quote)
+				.map(s -> String.format(PROPERTY_QUOTING_FORMAT, s))
+				.collect(Collectors.joining(","));
+		return String.format(STATEMENT_REL_FULLTEXT_IDX, name, type, key);
 	}
 
 	@Override

--- a/src/main/java/apoc/export/cypher/formatter/CypherFormatter.java
+++ b/src/main/java/apoc/export/cypher/formatter/CypherFormatter.java
@@ -2,9 +2,7 @@ package apoc.export.cypher.formatter;
 
 import apoc.export.util.ExportConfig;
 import apoc.export.util.Reporter;
-import org.neo4j.graphdb.GraphDatabaseService;
-import org.neo4j.graphdb.Node;
-import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.*;
 
 import java.io.PrintWriter;
 import java.util.Map;
@@ -22,6 +20,10 @@ public interface CypherFormatter {
 	String statementForRelationship(Relationship relationship, Map<String, Set<String>> uniqueConstraints, Set<String> indexedProperties);
 
 	String statementForIndex(String label, Iterable<String> keys);
+
+	String statementForNodeFullTextIndex(String name, Iterable<Label> labels, Iterable<String> keys);
+
+	String statementForRelationshipFullTextIndex(String name, Iterable<RelationshipType> types, Iterable<String> keys);
 
 	String statementForConstraint(String label, Iterable<String> keys);
 

--- a/src/main/java/apoc/export/util/ExportConfig.java
+++ b/src/main/java/apoc/export/util/ExportConfig.java
@@ -40,9 +40,11 @@ public class ExportConfig {
     private boolean separateHeader;
     private String arrayDelim;
     private Map<String, Object> optimizations;
+
     public enum OptimizationType {NONE, UNWIND_BATCH, UNWIND_BATCH_PARAMS}
     private OptimizationType optimizationType;
     private int unwindBatchSize;
+    private long awaitForIndexes;
 
     public int getBatchSize() {
         return batchSize;
@@ -98,6 +100,7 @@ public class ExportConfig {
         this.optimizationType = OptimizationType.valueOf(optimizations.getOrDefault("type", OptimizationType.UNWIND_BATCH.toString()).toString().toUpperCase());
         this.batchSize = ((Number)config.getOrDefault("batchSize", DEFAULT_BATCH_SIZE)).intValue();
         this.unwindBatchSize = ((Number)getOptimizations().getOrDefault("unwindBatchSize", DEFAULT_UNWIND_BATCH_SIZE)).intValue();
+        this.awaitForIndexes = ((Number)config.getOrDefault("awaitForIndexes", 300)).longValue();
         validate();
     }
 
@@ -186,6 +189,10 @@ public class ExportConfig {
 
     public OptimizationType getOptimizationType() {
         return optimizationType;
+    }
+
+    public long getAwaitForIndexes() {
+        return awaitForIndexes;
     }
 
 }

--- a/src/main/java/apoc/export/util/ExportFormat.java
+++ b/src/main/java/apoc/export/util/ExportFormat.java
@@ -1,7 +1,5 @@
 package apoc.export.util;
 
-import apoc.export.cypher.formatter.CypherFormatterUtils;
-
 import static java.lang.String.format;
 
 /**
@@ -15,7 +13,7 @@ public enum ExportFormat {
             format("COMMIT%n"), format("BEGIN%n"), format("SCHEMA AWAIT%n"), ""),
 
     CYPHER_SHELL("cypher-shell",
-            format(":commit%n"), format(":begin%n"), "", "CALL db.awaitIndex('%s(%s)');%n"),
+            format(":commit%n"), format(":begin%n"), "", "CALL db.awaitIndexes(%d);%n"),
 
     PLAIN_FORMAT("plain", "", "", "", ""),
 
@@ -63,7 +61,7 @@ public enum ExportFormat {
         return this.schemaAwait;
     }
 
-    public String indexAwait(String label, Iterable<String> properties){
-        return format(this.indexAwait, CypherFormatterUtils.label(label), CypherFormatterUtils.quote(properties));
+    public String indexAwait(long millis){
+        return format(this.indexAwait, millis);
     }
 }

--- a/src/test/java/apoc/export/cypher/ExportCypherTest.java
+++ b/src/test/java/apoc/export/cypher/ExportCypherTest.java
@@ -2,9 +2,11 @@ package apoc.export.cypher;
 
 import apoc.graph.Graphs;
 import apoc.util.TestUtil;
+import org.apache.commons.lang3.exception.ExceptionUtils;
 import org.junit.*;
 import org.junit.rules.TestName;
 import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.QueryExecutionException;
 import org.neo4j.graphdb.factory.GraphDatabaseSettings;
 import org.neo4j.test.TestGraphDatabaseFactory;
 
@@ -44,16 +46,13 @@ public class ExportCypherTest {
                 .setConfig("apoc.export.file.enabled", "true")
                 .newGraphDatabase();
         TestUtil.registerProcedure(db, ExportCypher.class, Graphs.class);
+        db.execute("CREATE INDEX ON :Bar(first_name, last_name)").close();
+        db.execute("CREATE INDEX ON :Foo(name)").close();
+        db.execute("CREATE CONSTRAINT ON (b:Bar) ASSERT b.name IS UNIQUE").close();
         if (testName.getMethodName().endsWith(OPTIMIZED)) {
-            db.execute("CREATE INDEX ON :Foo(name)").close();
-            db.execute("CREATE INDEX ON :Bar(first_name, last_name)").close();
-            db.execute("CREATE CONSTRAINT ON (b:Bar) ASSERT b.name IS UNIQUE").close();
             db.execute("CREATE (f:Foo {name:'foo', born:date('2018-10-31')})-[:KNOWS {since:2016}]->(b:Bar {name:'bar',age:42}),(c:Bar:Person {age:12}),(d:Bar {age:12})," +
                     " (t:Foo {name:'foo2', born:date('2017-09-29')})-[:KNOWS {since:2015}]->(e:Bar {name:'bar2',age:44}),({age:99})").close();
         } else if(testName.getMethodName().endsWith(ODD)) {
-            db.execute("CREATE INDEX ON :Foo(name)").close();
-            db.execute("CREATE INDEX ON :Bar(first_name, last_name)").close();
-            db.execute("CREATE CONSTRAINT ON (b:Bar) ASSERT b.name IS UNIQUE").close();
             db.execute("CREATE (f:Foo {name:'foo', born:date('2018-10-31')})," +
                     "(t:Foo {name:'foo2', born:date('2017-09-29')})," +
                     "(g:Foo {name:'foo3', born:date('2016-03-12')})," +
@@ -63,9 +62,6 @@ public class ExportCypherTest {
                     "(e:Bar {name:'bar2',age:44})," +
                     "(f)-[:KNOWS {since:2016}]->(b)").close();
         } else {
-            db.execute("CREATE INDEX ON :Foo(name)").close();
-            db.execute("CREATE INDEX ON :Bar(first_name, last_name)").close();
-            db.execute("CREATE CONSTRAINT ON (b:Bar) ASSERT b.name IS UNIQUE").close();
             db.execute("CREATE (f:Foo {name:'foo', born:date('2018-10-31')})-[:KNOWS {since:2016}]->(b:Bar {name:'bar',age:42}),(c:Bar {age:12})").close();
         }
     }
@@ -550,6 +546,67 @@ public class ExportCypherTest {
                 .replace(NEO4J_SHELL.schemaAwait(), CYPHER_SHELL.schemaAwait());
         assertEquals(expected, readFile(fileName));
     }
+    
+    @Test
+    public void exportMultiTokenIndex() {
+        // given
+        db.execute("CREATE (n:TempNode {value:'value'})");
+        db.execute("CREATE (n:TempNode2 {value:'value'})");
+        db.execute("CALL db.index.fulltext.createNodeIndex('MyCoolNodeFulltextIndex',['TempNode', 'TempNode2'],['value'])");
+
+        String query = "MATCH (t:TempNode) return t";
+        String file = null;
+        Map<String, Object> config = map("awaitForIndexes", 3000);
+        String expected = String.format(":begin%n" +
+                "CALL db.index.fulltext.createNodeIndex('MyCoolNodeFulltextIndex',['TempNode','TempNode2'],['value']);%n" +
+                "CREATE CONSTRAINT ON (node:`UNIQUE IMPORT LABEL`) ASSERT (node.`UNIQUE IMPORT ID`) IS UNIQUE;%n" +
+                ":commit%n" +
+                "CALL db.awaitIndexes(3000);%n" +
+                ":begin%n" +
+                "UNWIND [{_id:20, properties:{value:\"value\"}}] AS row%n" +
+                "CREATE (n:`UNIQUE IMPORT LABEL`{`UNIQUE IMPORT ID`: row._id}) SET n += row.properties SET n:TempNode;%n" +
+                ":commit%n" +
+                ":begin%n" +
+                "MATCH (n:`UNIQUE IMPORT LABEL`)  WITH n LIMIT 20000 REMOVE n:`UNIQUE IMPORT LABEL` REMOVE n.`UNIQUE IMPORT ID`;%n" +
+                ":commit%n" +
+                ":begin%n" +
+                "DROP CONSTRAINT ON (node:`UNIQUE IMPORT LABEL`) ASSERT (node.`UNIQUE IMPORT ID`) IS UNIQUE;%n" +
+                ":commit%n");
+
+        // when
+        TestUtil.testCall(db, "CALL apoc.export.cypher.query({query}, {file}, {config})",
+                map("query", query, "file", file, "config", config),
+                (r) -> {
+                    // then
+                    assertEquals(expected, r.get("cypherStatements"));
+                });
+    }
+
+    @Test(expected = QueryExecutionException.class)
+    public void shouldFailExportMultiTokenIndexForRelationship() {
+        // given
+        db.execute("CREATE (n:TempNode {value:'value'})");
+        db.execute("CREATE (n:TempNode2 {value:'value'})");
+        db.execute("CALL db.index.fulltext.createNodeIndex('MyCoolNodeFulltextIndex',['TempNode', 'TempNode2'],['value'])");
+
+        // TODO: We can't manage full-text rel indexes because of this bug: https://github.com/neo4j/neo4j/issues/12304
+        db.execute("CREATE (s:TempNode)-[:REL{rel_value: 'the rel value'}]->(e:TempNode2)");
+        db.execute("CALL db.index.fulltext.createRelationshipIndex('MyCoolRelFulltextIndex',['REL'],['rel_value'])");
+        String query = "MATCH (t:TempNode) return t";
+        String file = null;
+        Map<String, Object> config = map("awaitForIndexes", 3000);
+
+        try {
+            // when
+            TestUtil.testCall(db, "CALL apoc.export.cypher.query({query}, {file}, {config})",
+                    map("query", query, "file", file, "config", config),
+                    (r) -> {});
+        } catch (Exception e) {
+            String expected = "Full-text indexes on relationships are not supported, please delete them in order to complete the process";
+            assertEquals(expected, ExceptionUtils.getRootCause(e).getMessage());
+            throw e;
+        }
+    }
 
     private void assertResultsOptimized(String fileName, Map<String, Object> r) {
         assertEquals(7L, r.get("nodes"));
@@ -603,13 +660,9 @@ public class ExportCypherTest {
                 "COMMIT%n" +
                 "SCHEMA AWAIT%n");
 
-        public static final String EXPECTED_INDEXES_AWAIT = String.format("CALL db.awaitIndex(':Foo(name)');%n" +
-                "CALL db.awaitIndex(':Bar(first_name,last_name)');%n" +
-                "CALL db.awaitIndex(':Bar(name)');%n");
+        public static final String EXPECTED_INDEXES_AWAIT = String.format("CALL db.awaitIndexes(300);%n");
 
-        private static final String EXPECTED_INDEXES_AWAIT_QUERY = String.format("CALL db.awaitIndex(':Foo(name)');%n" +
-                "CALL db.awaitIndex(':Bar(name)');%n" +
-                "CALL db.awaitIndex(':Bar(first_name,last_name)');%n");
+        private static final String EXPECTED_INDEXES_AWAIT_QUERY = String.format("CALL db.awaitIndex(300);%n");
 
         static final String EXPECTED_RELATIONSHIPS = String.format("BEGIN%n" +
                 "MATCH (n1:`UNIQUE IMPORT LABEL`{`UNIQUE IMPORT ID`:0}), (n2:Bar{name:\"bar\"}) CREATE (n1)-[r:KNOWS {since:2016}]->(n2);%n" +

--- a/src/test/java/apoc/load/LoadS3Test.java
+++ b/src/test/java/apoc/load/LoadS3Test.java
@@ -14,6 +14,7 @@ import static apoc.util.TestUtil.testResult;
 import static java.util.Arrays.asList;
 import static org.junit.Assert.assertEquals;
 
+@Ignore("investigate minio error or move the test to Localstack")
 public class LoadS3Test {
 
     private GraphDatabaseService db;

